src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -25,26 +27,59 @@
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
 import java.time.LocalDate;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.json.JSONException;
+import org.json.JSONObject;
 
 /**
- * Servlet that schedules tasks on tomorrow.
+ * Schedules tasks on tomorrow.
  */
 @WebServlet("/schedule")
 public class ScheduleServlet extends HttpServlet {
+
+  public static final String TASK_ID_LIST_KEY = "taskId";
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    TasksProvider taskProvider = new TasksProvider();
+    List<Task> tasks = taskProvider.getTasks();
+    if(!request.getParameterMap().containsKey(TASK_ID_LIST_KEY)){
+      sendJsonResponse(response, "Select some tasks to schedule.");
+      return;
+    }
+    Set<String> selectedIds = Arrays.stream(request.getParameterValues(TASK_ID_LIST_KEY)).collect(Collectors.toSet());
+    List<String> titlesOfTasksToSchedule = filterSelectedTaskTitles(selectedIds, tasks);
+
     CalendarInterface calendarInterface = new CalendarInterface();
     List<Event> calendarEvents = calendarInterface.loadPrimaryCalendarEventsOfTomorrow();
     String timeZone = calendarInterface.getPrimaryCalendarTimeZone();
     LocalDate tomorrow = calendarInterface.getUsersTomorrowStart().toLocalDate();
-    List<Event> tasksEvent = Scheduler.schedule(calendarEvents, timeZone, tomorrow);
+
+
+    List<Event> tasksEvent = Scheduler.schedule(calendarEvents, titlesOfTasksToSchedule, timeZone, tomorrow);
     for (Event event : tasksEvent) {
       calendarInterface.insertEventToPrimary(event);
     }
 
-    response.setContentType(MediaType.TEXT_PLAIN);
+    sendJsonResponse(response, tasksEvent.size() + " tasks inserted on " + tomorrow);
+  }
+
+  private void sendJsonResponse(HttpServletResponse response, String responseMessage) throws IOException {
```

Comment:
Wasn't some of this code already in Stefano's PR? Maybe check on that with him to avoid conflicts.

Reply:
No, I implemented the JSON response in the `UserDataServlet`.